src/CommandEvent.cc

```diff
@@ -77,106 +76,152 @@ CommandEvent::CommandEvent() : Event() {
+/**
+  * \brief Constructor for CommandEvent
+  * \param originator originating location of this event
+  * \param destination destination location for this event
+  * \param ps PropertySet to pass in this event
+  * \param filterable additional, broker-filterable, PropertySet parameters
+  */
+CommandEvent::CommandEvent(const LocationID&  originator, const LocationID&  destination, const PropertySet& ps, const PropertySet& filterable) : Event(ps, filterable) {
+    _constructor(originator, destination);
 }
 
-void CommandEvent::_constructor( const std::string& runId, const int64_t originatorId, const int64_t destinationId, const PropertySet& ps) {
-    _init();
+/**
+  * \brief Constructor for CommandEvent
+  * \param runId name of the run which this event is used in
+  * \param originator originating location of this event
+  * \param destination destination location for this event
+  * \param psp PropertySet to pass in this event
+  */
+CommandEvent::CommandEvent( const std::string& runId, const LocationID&  originator, const LocationID& destination, const PropertySet::Ptr psp) : Event(runId, *psp) {
+    _constructor(originator, destination);
+}
 
-    EventSystem eventSystem = EventSystem().getDefaultEventSystem();
+/**
+  * \brief Constructor for CommandEvent
+  * \param runId name of the run which this event is used in
+  * \param originator originating location of this event
+  * \param destination destination location for this event
+  * \param ps PropertySet to pass in this event
+  */
+CommandEvent::CommandEvent( const std::string& runId, const LocationID&  originator, const LocationID&  destination, const PropertySet& ps) : Event(runId, ps) {
+    _constructor(originator, destination);
+}
 
-    //_originatorId = eventSystem.createOriginatorId();
-    _psp->set(ORIGINATORID, originatorId);
-    _psp->set(ORIG_LOCALID, eventSystem.extractLocalId(originatorId));
-    _psp->set(ORIG_PROCESSID, eventSystem.extractProcessId(originatorId));
-    _psp->set(ORIG_IPID, eventSystem.extractIPId(originatorId));
+/**
+  * \brief Constructor for CommandEvent
+  * \param runId name of the run which this event is used in
+  * \param originator originating location of this event
+  * \param destination destination location for this event
+  * \param ps PropertySet to pass in this event
+  * \param filterable additional, broker-filterable, PropertySet parameters
+  */
+CommandEvent::CommandEvent( const std::string& runId, const LocationID&  originator, const LocationID&  destination, const PropertySet& ps, const PropertySet& filterable) : Event(runId, ps, filterable) {
+    _constructor(originator, destination);
+}
+
+/** private method common to all constructors containing, originator, the 
+  * originating location of this event, and destination, the destination
+  * location for this event.
+  */
+void CommandEvent::_constructor(const LocationID&  originator, const LocationID&  destination) {
```

Comment:
Is there a better name than just "constructor"?  It's not really constructing.  It is initializing, so maybe overload `_init`?